repub.go

```diff
@@ -13,36 +14,35 @@ type PubFunc func(context.Context, cid.Cid) error
 
 // Republisher manages when to publish a given entry.
 type Republisher struct {
-	TimeoutLong  time.Duration
-	TimeoutShort time.Duration
-	RetryTimeout time.Duration
-	pubfunc      PubFunc
+	pubfunc PubFunc
 
 	update           chan cid.Cid
 	immediatePublish chan chan struct{}
 
-	ctx    context.Context
-	cancel func()
+	cancel  func()
+	once    sync.Once
+	stopped chan struct{}
 }
 
 // NewRepublisher creates a new Republisher object to republish the given root
-// using the given short and long time intervals.
-func NewRepublisher(ctx context.Context, pf PubFunc, tshort, tlong time.Duration) *Republisher {
+// using the short and long time intervals.
+func NewRepublisher(ctx context.Context, pf PubFunc, tshort, tlong time.Duration, lastPublished cid.Cid) *Republisher {
 	ctx, cancel := context.WithCancel(ctx)
-	return &Republisher{
-		TimeoutShort:     tshort,
-		TimeoutLong:      tlong,
-		RetryTimeout:     tlong,
+	rp := &Republisher{
 		update:           make(chan cid.Cid, 1),
 		pubfunc:          pf,
 		immediatePublish: make(chan chan struct{}),
-		ctx:              ctx,
 		cancel:           cancel,
+		stopped:          make(chan struct{}),
```

Comment:
This was actually more straightforward than I originally thought, but still it would be useful to add a concise comment explaining this channel will be closed after the main `run` goroutine ends (considering now that `run` is triggered internally only once and not by the client).

Reply:
Added comment in type definition.